audioipc/src/shm.rs

```diff
@@ -7,22 +7,96 @@ use crate::errors::*;
 use memmap::{Mmap, MmapMut, MmapOptions};
 use std::cell::UnsafeCell;
 use std::fs::{remove_file, File, OpenOptions};
-use std::path::Path;
+use std::path::PathBuf;
 use std::sync::{atomic, Arc};
+use std::convert::TryInto;
+use std::env::temp_dir;
+
+fn open_shm_file(id: &str) -> Result<File> {
+    if cfg!(target_os = "linux") {
+        let mut path = PathBuf::from("/dev/shm");
+        path.push(id);
+
+        if let Ok(file) = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create_new(true)
+            .open(&path) {
+                let _ = remove_file(&path);
```

Comment:
Why the file is opened but removed immediately? The original implementation does the same thing but I don't understand. 

According to the [document](https://doc.rust-lang.org/std/fs/fn.remove_file.html),
>  there is no guarantee that the file is immediately deleted

Is the something that needs to be handled?

Reply:
The files we create are temporary and exist only to give the shared memory area a size and a way to reference it (e.g. an fd).  The current portable approach is to create a file and immediately remove it.  This works reasonably well on Unix-like systems, but Windows (for example) does have the issue where the temporary file may not be removed immediately.  We use unique filenames and per-user (or more restricted) temporary directories to mitigate the risk of the file being visible longer than necessary.

Ideally, we'd be able to create anonymous sharable mapping, which don't have a public addressable name at all, but can still be shared between processes by reference (e.g. fd or HANDLE).  On Linux, we can do that with memfd, but that's non-portable and other platforms require different approaches or don't offer anything better than our current temporary file approach.